src/viewers/VivViewer.js

```diff
@@ -78,13 +89,56 @@ export default class VivViewer extends PureComponent {
     return viewState;
   }
 
+  componentDidUpdate(prevProps) {
+    const { props } = this;
+    const { views } = props;
+    // Only update state if the previous viewState prop does not match the current one
+    // so that people can update viewState
+    const changed = views
+      .filter(view => {
+        const prevViewState = prevProps.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        const currViewState = props.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        return (
+          currViewState &&
+          (!prevViewState || !areViewStatesEqual(currViewState, prevViewState))
```

Comment:
The test for `!prevViewState` isn't needed. However, the test for `curViewState` is needed because otherwise the following may return `undefined`:
```
const viewState = props.viewStates.find(v => v.id === view.id);
```
One could short-circuit `filter` slightly and not run `prevProps.viewStates.find()` if `!currViewState`.